### PR TITLE
Implemented Confectionary baking scenario

### PR DESCRIPTION
**Pull request by wolmanr**
I chose the confectionary baking scenario. I did the work, please let me know if it needs any changes

### CODE REVIEW

File: 4 Reports.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    Winter: November - April
+    Spring: May and June*/
+
+select Season = case
+                when datepart(month,b.OrderDate) in (7, 8) then 'Summer'
+                when datepart(month,b.OrderDate) in (9, 10) then 'Holiday time'
+                when datepart(month,b.OrderDate) in (11, 12, 1, 2, 3, 4) then 'Winter'
+                when datepart(month,b.OrderDate) in (5, 6) then 'Spring'
+                end,
+b.StoreBranch, b.StoreBranch , NumOfOrders = count(ItemType)
```
Comment (astrohlicpu) [inline 4 Reports.sql:27]: You repeated StoreBranch twice and forgot to include OrderOccasion. Probably a typo :)

**Review by astrohlicpu — CHANGES_REQUESTED** [review]
Amazing job! See some small comments

File: 1 Table.sql [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            check(BaseType in ('Chocolate', 'Vanilla', 'Coconut', 'Chocolate Peanut Butter', 'Banana', 'Strawberry shortcake', 'sugar')),
+
+    ItemType varchar(30) not null
+        constraint ck_Baking_item_type_must_be_either_cake_or_cookie_or_cupcake check(ItemType in ('cake', 'cookie', 'cupcake')),
+    
+    Topping varchar(50) not null
+        constraint ck_Baking_topping_must_be_either_icing_fondant_choc_caramel_strawberry_coconut_peanut_butter_vanilla_or_no_topping
+            check (Topping in ('Royal icing', 'fondant', 'frosting', 'chocolate', 'caramel', 'strawberry', 'coconut', 'peanut butter', 'vanilla', 'no topping')),
+
+    Picture varchar(3) not null
```
[the remark this answers is not in the document]
Comment (wolmanr) [reply 1 Table.sql:28]: I changed it to a bit column

File: 1 Table.sql [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            WHEN ItemType = 'cupcake' THEN 
+            (3 + CASE WHEN Picture = 'yes' THEN 0 ELSE 0 END) 
+            ELSE 0 
+            END) persisted,
+
+     OrderPrice AS (Amount * 
+            CASE
+            WHEN ItemType = 'cookie' THEN
+                (3.50 + CASE WHEN Picture = 'yes' THEN 1.50 ELSE 0 END)
+            WHEN ItemType = 'cupcake' THEN
```
[the remark this answers is not in the document]
Comment (wolmanr) [reply 1 Table.sql]: I changed this

File: 4 Reports.sql (hunk 1 above)
Comment (wolmanr) [reply 4 Reports.sql]: Whoops! I fixed that

**Review by wolmanr — COMMENTED** [review]
Thanks for the feedback. I addressed all the issues you raised. See my latest comments.

File: 1 Table.sql [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ 
+    PricePerItem AS
+        (CASE 
+            WHEN BaseType = 'Strawberry shortcake' THEN 
+            (55 + CASE WHEN Picture = 'yes' THEN 8 ELSE 0 END)  
+            WHEN ItemType = 'cake' THEN 
+            (50 + CASE WHEN Picture = 'yes' THEN 8 ELSE 0 END) 
+            WHEN ItemType = 'cookie' THEN 
+            (3.50 + CASE WHEN Picture = 'yes' THEN 1.50 ELSE 0 END)  
+            WHEN ItemType = 'cupcake' THEN
```
[the remark this answers is not in the document]
Comment (wolmanr) [reply 1 Table.sql]: I changed this

File: 1 Table.sql [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        (CASE 
+            WHEN BaseType = 'Strawberry shortcake' THEN 
+            (55 + CASE WHEN Picture = 'yes' THEN 8 ELSE 0 END)  
+            WHEN ItemType = 'cake' THEN 
+            (50 + CASE WHEN Picture = 'yes' THEN 8 ELSE 0 END) 
+            WHEN ItemType = 'cookie' THEN 
+            (3.50 + CASE WHEN Picture = 'yes' THEN 1.50 ELSE 0 END)  
+            WHEN ItemType = 'cupcake' THEN 
+            (3 + CASE WHEN Picture = 'yes' THEN 0 ELSE 0 END) 
+            ELSE 0
```
[the remark this answers is not in the document]
Comment (wolmanr) [reply 1 Table.sql]: I took away the else

File: 1 Table.sql [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        constraint ck_Baking_item_type_must_be_either_cake_or_cookie_or_cupcake check(ItemType in ('cake', 'cookie', 'cupcake')),
+    
+    Topping varchar(50) not null
+        constraint ck_Baking_topping_must_be_either_icing_fondant_choc_caramel_strawberry_coconut_peanut_butter_vanilla_or_no_topping
+            check (Topping in ('Royal icing', 'fondant', 'frosting', 'chocolate', 'caramel', 'strawberry', 'coconut', 'peanut butter', 'vanilla', 'no topping')),
+
+    Picture varchar(3) not null
+        constraint ck_Baking_picture_must_be_either_yes_or_no check(picture in ('yes', 'no')),
+
+    OrderSpecifics varchar(100) not null
```
[the remark this answers is not in the document]
Comment (wolmanr) [reply 1 Table.sql]: I fixed this

**Comment by wolmanr** [comment]
Please check if I fixed everything correctly. Thank you

File: 1 Table.sql (hunk 6 above)
[the remark this answers is not in the document]
Comment (astrohlicpu) [reply 1 Table.sql]: But you still don't want a null, only a blank

**Review by astrohlicpu — CHANGES_REQUESTED** [review]
Please see comments.
I also wanted to point out, that the file Database is misspelled and that you numbered your files 0134, skipping 2.

File: 1 Table.sql (hunk 6 above)
[the remark this answers is not in the document]
Comment (wolmanr) [reply 1 Table.sql]: I am not understanding, I made it not null.

**Comment by wolmanr** [comment]
I changed the names of the incorrect file names. I did not understand what was wrong with the picture column. If you can elaborate on that, maybe I can fix it.

File: 1 Table.sql (hunk 6 above)
[the remark this answers is not in the document]
Comment (astrohlicpu) [reply 1 Table.sql]: It shows here as not null, but in the latest push it allows nulls. I am going to close this 2 conversations, and comment on the latest files.

File: 1 Table.sql [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    ItemType varchar(30) not null
+        constraint ck_Baking_item_type_must_be_either_cake_or_cookie_or_cupcake check(ItemType in ('cake', 'cookie', 'cupcake')),
+    
+    Topping varchar(50) not null
+        constraint ck_Baking_topping_must_be_either_icing_fondant_choc_caramel_strawberry_coconut_peanut_butter_vanilla_or_no_topping
+            check (Topping in ('Royal icing', 'fondant', 'frosting', 'chocolate', 'caramel', 'strawberry', 'coconut', 'peanut butter', 'vanilla', 'no topping')),
+
+    Picture bit not null,
+
+    OrderSpecifics varchar(100),
```
Comment (astrohlicpu) [inline 1 Table.sql:30]: The column OrderSpecifics is not marked as not null. Even though we might leave it blank, we don't need it to be null.
The picture column is fine, my mistake.

**Review by astrohlicpu — COMMENTED** [review]
Everything looks great, just this last null. I am already going to close the HW issue, no need to resubmit it after its done.

File: 1 Table.sql (hunk 7 above)
Comment (astrohlicpu) [reply 1 Table.sql:30]: Yes, you can close it.

**Pull request closed without merging**